fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java

```diff
@@ -114,35 +117,39 @@ public Void visitCreateMaterializedViewStatement(CreateMaterializedViewStatement
             // convert queryStatement to sql and set
             statement.setInlineViewDef(ViewDefBuilder.build(queryStatement));
             // collect table from query statement
-            Map<TableName, Table> tableNameTableMap = AnalyzerUtils.collectAllTableAndViewWithAlias(queryStatement);
-            Set<Long> baseTableIds = Sets.newHashSet();
+            Map<TableName, Table> tableNameTableMap = AnalyzerUtils.collectAllTable(queryStatement);
+            List<MaterializedView.BaseTableInfo> baseTableInfos = Lists.newArrayList();
             Database db = context.getGlobalStateMgr().getDb(statement.getTableName().getDb());
             if (db == null) {
                 throw new SemanticException("Can not find database:" + statement.getTableName().getDb());
             }
             if (tableNameTableMap.isEmpty()) {
                 throw new SemanticException("Can not find base table in query statement");
             }
-            tableNameTableMap.values().forEach(table -> {
-                if (db.getTable(table.getId()) == null) {
-                    throw new SemanticException(
-                            "Materialized view do not support table: " + table.getName() +
-                                    " do not exist in database: " + db.getOriginName());
-                }
-                if (!(table instanceof OlapTable)) {
-                    throw new SemanticException(
-                            "Materialized view only supports olap table, but the type of table: " +
-                                    table.getName() + " is: " + table.getType().name());
+            tableNameTableMap.forEach((tableNameInfo, table) -> {
+                if (table == null) {
+                    throw new SemanticException("Materialized view do not support table: " + tableNameInfo.getTbl() +
```

Comment:
How about simplify the message as 'referenced table not exist'

Reply:
ok， I will simplify the message in next PR